### PR TITLE
fix pulling single file in a directory

### PR DESCRIPTION
**Pull request by horace-stavropoulos**
When pulling, and there's only one signed url:
  checking if the signed url matches the given name

### CODE REVIEW

**Comment by AleksandarCole** [comment]
Corresponding ticket: https://github.com/renderedtext/issues/issues/2636

File: pkg/gcs/gcs.go [hunk 1]
```diff
@@ -356,6 +356,17 @@ func PullPaths(dst, src string) (string, string) {
 	return newDst, newSrc
 }
 
+func strIndexByteNthAfter(s string, b byte, count int) string {
```
Comment (shiroyasha) [inline pkg/gcs/gcs.go:359]: The name of this function is not clear. What is the job of this function?

File: pkg/gcs/gcs.go [hunk 2]
```diff
@@ -371,17 +382,19 @@ func PullGCS(dst, src string, force bool) (ok bool) {
...
 	}
 	if len(t.Urls) == 1 {
-		if ok = PullFileGCS(dst, t.Urls[0].URL, force); !ok {
+		url := t.Urls[0].URL
+		obj := ParseURL(url)
+		obj = strIndexByteNthAfter(obj, '/', 3) // removing <project-name>/<level>/<projectID>/
+		if obj == src {                         // they are the same: requested single file pull
+			return PullFileGCS(dst, url, force)
+		} // otherwise it will be downloaded as a directory
+	}
```
Comment (shiroyasha) [inline pkg/gcs/gcs.go:391]: The complexity of the PullGCS function is too high, please extract the logic from this if to a dedicated function.

**Review by shiroyasha — CHANGES_REQUESTED** [review]
The initial code looks good. We need two things before we can proceed:

- Cover this behavior with tests.
- The new code increases the cognitive load of the PullGCS function. Some slight refactoring is needed.

**Review by shiroyasha — APPROVED** [review]

**Pull request merged**